tests/java/TimeDeltaTests.java

```diff
@@ -0,0 +1,564 @@
+import org.python.Object;
+import org.python.exceptions.OverflowError;
+import org.python.exceptions.TypeError;
+import org.python.exceptions.ValueError;
+import org.python.stdlib.datetime.TimeDelta;
+
+import org.junit.Test;
+import org.python.types.*;
+import org.python.types.Float;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.*;
+
+public class TimeDeltaTests {
+    public static TimeDelta createDelta(double days, double seconds, double microseconds, double milliseconds,
+                                        double minutes, double hours, double weeks) {
+        Map<String, Object> kwargs = new HashMap<>();
+        kwargs.put("days", new Float(days));
+        kwargs.put("seconds", new Float(seconds));
+        kwargs.put("microseconds", new Float(microseconds));
+        kwargs.put("milliseconds", new Float(milliseconds));
+        kwargs.put("minutes", new Float(minutes));
+        kwargs.put("hours", new Float(hours));
+        kwargs.put("weeks", new Float(weeks));
+        return new TimeDelta(TimeDelta.EMPTY_ARGS, kwargs);
+    }
+
+    public static void assertDelta(long days, long seconds, long microseconds, TimeDelta delta) {
+        assertEquals(days, ((Int) delta.days).value);
+        assertEquals(seconds, ((Int) delta.seconds).value);
+        assertEquals(microseconds, ((Int) delta.microseconds).value);
+    }
+
+    public static void assertBool(boolean expected, Object bool) {
+        assertEquals(expected, ((Bool) bool).value);
+    }
+
+    @Test
+    public void When_InstantiateWithAllArgs_Expect_ValidTimeDelta() {
+        Object[] args = {
+            Int.getInt(1),
+            Int.getInt(2),
+            Int.getInt(3),
+            Int.getInt(4),
+            Int.getInt(5),
+            Int.getInt(6),
+            Int.getInt(7),
+        };
+
+        TimeDelta delta = new TimeDelta(args, TimeDelta.EMPTY_KWARGS);
+
+        assertDelta(50, 21902, 4003, delta);
+    }
+
+    @Test
+    public void When_InstantiateWithAllKwargs_Expect_ValidTimeDelta() {
```

Comment:
Why do some methods start with `When_` and some with `Test_`? Is there some system we don't understand?

Reply:
No system for that no, coherent naming will be fixed